### PR TITLE
add unit tests for go type gen function

### PR DESCRIPTION
**Pull request by sonasingh46**
This PR adds unit test for `GoTypeDef()` method.

### CODE REVIEW

File: pkg/codegen/schema_test.go [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+			},
+			want: "*int",
+		},
+
+		// When field is write only, it will always be pointer unless pointer is
+		// skipped by setting SkipOptionalPointer flag
+		{
+			name: "When field is write only",
+			fields: fields{
+				GlobalStateDisableRequiredReadOnlyAsPointer: true,
```
Comment (jamietanna) [inline pkg/codegen/schema_test.go:179]: are we missing a case for the case where this is set to `false`?

File: pkg/codegen/schema_test.go [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+		t.Run(tt.name, func(t *testing.T) {
+			globalState.options.Compatibility.DisableRequiredReadOnlyAsPointer = tt.fields.GlobalStateDisableRequiredReadOnlyAsPointer
+			p := Property{
+				Schema:    tt.fields.Schema,
+				Required:  tt.fields.Required,
+				Nullable:  tt.fields.Nullable,
+				ReadOnly:  tt.fields.ReadOnly,
+				WriteOnly: tt.fields.WriteOnly,
+			}
+			assert.Equalf(t, tt.want, p.GoTypeDef(), "GoTypeDef()")
```
Comment (jamietanna) [inline pkg/codegen/schema_test.go:199]:
```suggestion
			assert.Equal(t, tt.want, p.GoTypeDef())
```

**Comment by jamietanna** [comment]
Couple of things to address, then LGTM

File: pkg/codegen/schema_test.go (hunk 1 above)
Comment (sonasingh46) [reply pkg/codegen/schema_test.go:179]: added...

File: pkg/codegen/schema_test.go [hunk 3]
```diff
@@ -174,7 +174,7 @@ func TestProperty_GoTypeDef(t *testing.T) {
 		// When field is write only, it will always be pointer unless pointer is
 		// skipped by setting SkipOptionalPointer flag
 		{
-			name: "When field is write only",
+			name: "When field is write only and read only pointer disabled",
```
Comment (jamietanna) [inline pkg/codegen/schema_test.go:177]: is the name of this test right, when the flag is `GlobalStateDisableRequiredReadOnlyAsPointer: true`? Or should we rename the tests?

Comment (sonasingh46) [reply pkg/codegen/schema_test.go:177]: I did not understand what you mean. 
Are you saying rename to the following: 
```
"When field is write only and the flag is GlobalStateDisableRequiredReadOnlyAsPointer: true
```

What is incorrect here?

Comment (jamietanna) [reply pkg/codegen/schema_test.go:177]: Ignore me, I'd misread

**Review by jamietanna — APPROVED** [review]

**Pull request merged**